# Random Forest Model

# Importing dataset and libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df=pd.read_csv(r'C:\Users\prith\Downloads\dataset_full.csv')

In [3]:
df.head()

qty_dot_url  qty_hyphen_url  qty_underline_url  qty_slash_url  \
0            3               0                  0              1   
1            5               0                  1              3   
2            2               0                  0              1   
3            4               0                  2              5   
4            2               0                  0              0   

   qty_questionmark_url  qty_equal_url  qty_at_url  qty_and_url  \
0                     0              0           0            0   
1                     0              3           0            2   
2                     0              0           0            0   
3                     0              0           0            0   
4                     0              0           0            0   

   qty_exclamation_url  qty_space_url  ...  qty_ip_resolved  qty_nameservers  \
0                    0              0  ...                1                2   
1                    0              0  ...                1                2   
2                    0              0  ...                1                2   
3                    0              0  ...                1                2   
4                    0              0  ...                1                2   

   qty_mx_servers  ttl_hostname  tls_ssl_certificate  qty_redirects  \
0               0           892                    0              0   
1               1          9540                    1              0   
2               3           589                    1              0   
3               0           292                    1              0   
4               1          3597                    0              1   

   url_google_index  domain_google_index  url_shortened  phishing  
0                 0                    0              0         1  
1                 0                    0              0         1  
2                 0                    0              0         0  
3                 0                    0              0         1  
4                 0                    0              0         0  

[5 rows x 112 columns]

# Splitting into Training and Test set

In [4]:
X=df.drop('phishing',axis=1)
y=df['phishing']

In [5]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [6]:
from sklearn.preprocessing import StandardScaler
scaler=StandardScaler()
X_train=scaler.fit_transform(X_train)
X_test=scaler.transform(X_test)

# Creating Random Forest Model

In [7]:
from sklearn.ensemble import RandomForestClassifier

In [17]:
rf=RandomForestClassifier(n_estimators=150)

In [18]:
rf.fit(X_train,y_train)

RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
                       max_depth=None, max_features='auto', max_leaf_nodes=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=150,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

# Model Prediction

In [19]:
pred=rf.predict(X_test)

In [11]:
from sklearn.metrics import confusion_matrix,classification_report

In [20]:
print(classification_report(y_test,pred))

              precision    recall  f1-score   support

           0       0.98      0.97      0.98     17449
           1       0.95      0.96      0.96      9146

    accuracy                           0.97     26595
   macro avg       0.96      0.97      0.97     26595
weighted avg       0.97      0.97      0.97     26595



# Accuracy of 97% achieved using 150 estimators in Random Forests